storage/dataflux/fast_list.go

```diff
@@ -53,18 +59,25 @@ type Input struct {
 type Lister struct {
 	// method indicates the listing method(open, sequential, worksteal) to be used for listing.
 	method listingMethod
+
 	// pageToken is the token to use for sequential listing.
 	pageToken string
+
 	// ranges is the channel to store the start and end ranges to be listed by the workers in worksteal listing.
 	ranges chan *listRange
```

Comment:
Let's leave this out for now since it is unused. Some of the code where you're looking at the length of the channel in fast_list.go looks questionable to me.

Reply:
Removed the line for now.

Listing is paused when we list given batch size objects. ranges is used to continue listing from where it was left off when NextBatch is called again.
listRange stores start and end range of each worker in worksteal algorithm. When NextBatch is called again, listRange is used by all the workers to find start and end ranges. 

I am checking length of the channel to make sure all the ranges have been listed.